File: AdventOfCode2020/Day2.cpp
#include <iostream>
#include <fstream>

using namespace std;


int main()
{
	int lowBound, highBound;
	char letter, trash;
	string input;
	ifstream infile;
	infile.open("input.txt");
	int valid = 0;
	while (!infile.eof())
	{
	
		//6-7 z: dqzzzjbzz
		//13 - 16 j: jjjvjmjjkjjjjjjj
		//int counter = 0;
		infile >> lowBound  >> trash >> highBound >>  letter >> trash >> input;
		//cout << lowBound << " " << highBound <<" " <<letter << " "<< input;
		if ((input[lowBound-1] == letter) ^ (input[highBound-1] == letter))
		{
			valid++;
		}
	}
	cout << valid;
}

/*for (int i = 0; i < input.size(); i++)
		{
			if (input[i] == letter)
			{
				counter++;
			}
		}
		if (counter >= lowBound && counter <= highBound)
		{
			valid++;
		}*/

File: AdventOfCode2020/Day2.py

def main():
    with open('day2.txt') as f:
        lines=[]
        lines=f.read().splitlines()
        part1(lines)
        part2(lines)

def part1(lines):
    counter=0
        #print (len(lines))
    for line in lines:
            #6-7 z: dqzzzjbzz
        lowBound=(int(line.split('-')[0]))
        highBound=(int(line.split('-')[1].split(' ')[0]))
        key=(line.split(' ')[1][0])
        inputs=(line.split(" ")[2])
        if(inputs.count(key)<=highBound and inputs.count(key)>=lowBound):
            counter+=1
    print(counter)
        #print(highBound)
        #print(lowBound)
        #print(key)

def part2(lines):
    counter=0
    for line in lines:
            #6-7 z: dqzzzjbzz
        lowBound=(int(line.split('-')[0]))
        highBound=(int(line.split('-')[1].split(' ')[0]))
        key=(line.split(' ')[1][0])
        inputs=(line.split(" ")[2])
        if((inputs[(lowBound-1)]==key) ^ (inputs[(highBound-1)]==key)):
            counter+=1
    print(counter)

if __name__ == "__main__":
    main()